Eslint fails with cannot find module error

### Summary

<!-- Begin your issue with a one sentence summary -->
Eslint doesn't work at all with the following error

```
Language server eslint:

initializing server eslint, id 16
-- stderr--
node:internal/modules/cjs/loader:1408
  throw err;
  ^

Error: Cannot find module '/home/yagiz/.local/share/zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server/out/eslintServer.js'
    at Function._resolveFilename (node:internal/modules/cjs/loader:1405:15)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1061:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1066:22)
    at Function._load (node:internal/modules/cjs/loader:1215:37)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:151:5)
    at node:internal/main/run_main_module:33:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v23.11.0
```

Removing `/home/yagiz/.local/share/zed/languages/eslint` and restarting Zed reinstalls the eslint language scripts but doesn't fix this problem.

### Zed Version and System Specs

Zed: v0.194.3 (Zed) 
OS: Linux Wayland ubuntu 24.04
Memory: 250.9 GiB
Architecture: x86_64
GPU: AMD Radeon PRO W6400 || AMD open-source driver || 2024.Q3.3 (LLPC)

--- a/crates/languages/src/typescript.rs
+++ b/crates/languages/src/typescript.rs
@@ -910,7 +910,7 @@ impl LspAdapter for EsLintLspAdapter {
         let server_path = destination_path.join(Self::SERVER_PATH);
 
         if fs::metadata(&server_path).await.is_err() {
-            remove_matching(&container_dir, |entry| entry != destination_path).await;
+            remove_matching(&container_dir, |_| true).await;
 
             download_server_binary(
                 delegate,

